scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1289,13 +1315,21 @@ private void handleTimeout(DTLSFlight flight) {
 			LOGGER.log(Level.FINE, "Re-transmitting flight for [{0}], [{1}] retransmissions left",
 					new Object[]{flight.getPeerAddress(), max - flight.getTries() - 1});
 
-			flight.incrementTries();
-			sendFlight(flight);
+			try {
+				flight.incrementTries();
+				flight.setNewSequenceNumbers();
```

Comment:
Now `setNewSequenceNumbers()` is only called in `handleTimeout`. Before this was called on `sendFlight`. So It seems there is a different behavior.  Am I wrong ?

(Ok I reread the spec and I mixed `message_seq`  and `record.sequence_number`, I agree with you now)